### PR TITLE
s4584245-Yue-GCN

### PR DESCRIPTION
**Pull request by yuexiangzh**
Hi Shakes,
This pull request contains the files relating to implementing the GCN task. The descriptions of the files are below:
GCN_Layer.py: Consrtucting the two GCN layers.
load_data.py: Process the data-set.
test_driver_script.py: It seems like the main function in this project.
tsne.png：A TSNE embeddings plot with ground truth in colors.
acc.png: The line chart of Validation accuracy.
gcn.png: It would be used in README.md file.
README.md：The introduction of project and result(accuracy&tsne) of GCNs
Download the files in the same document and run the test_driver_script.py.:)

Best Regards,
YueXiang Zhao
s4584245

### CODE REVIEW

**Comment by gayanku** [comment]
- Last commit on Nov 1. Commits span 4 days(Oct 28 to Nov 1)
- Commit comments could be more descriptive.
- Readme is well done with background, results, diagrams.
- multi layer (2) layer GCN, 
**- invalid for due to code being incomplete and not matching provided results.**

- Reported a _val_ accuracy (+80%). This is however not from the code in the PR.
- Unsure if the code runs. test_driver_script.py calls a non-existent method train()
- Low code structure and modularization. Avoid the use of global variables. For example tsne() 
- Has used fully supervised split (10000/2000/3000). In future make sure that n_train < n_test for Semi supervised.

**Comment by yuexiangzh** [comment]
> * 最后一次提交时间为 11 月 1 日。提交时间跨度为 4 天（10 月 28 日至 11 月 1 日）
> * 提交评论可能更具描述性。
> * 自述文件在背景、结果、图表方面做得很好。
> * 多层 (2) 层 GCN，
>   **- 由于代码不完整且与提供的结果不匹配而无效。**
> * 报告了_val_准确度 (+80%)。然而，这不是来自 PR 中的代码。
> * 不确定代码是否运行。test_driver_script.py 调用了一个不存在的方法 train()
> * 低代码结构和模块化。避免使用全局变量。例如 tsne()
> * 已使用完全监督拆分 (10000/2000/3000)。将来确保 n_train < n_test 用于半监督。

**Comment by yuexiangzh** [comment]
Hello gayanku，thanks for your comments, I have fixed the bugs in the test_drive_script.I did not clear the Class Train() so that you cannot call method train.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by yuexiangzh** [comment]
I can run the  test_drive_script.py in my pc, but I did not publish the latest version of code. Now I have fixed all bugs and republished it to the pull request.
1. Spliting the dataset into three parts (test:10000/valid:2000/train:3000) , so that n_train < n_test because of Semi supervised(load_data.py)
2. Fixed the naming error in GCN_layer.py
3. Clear Train() class in test_driver_script.py
4. Method train() has defined in GCN_train.py(Class Train).
As a result, the val_accuracy will reach at about 88%. Moveover, I will update the result and README.md.

**Comment by yuexiangzh** [comment]
You are able to run the test_driver_script.py now, here is a screenshot of the running. :)

U$BOVE96SH}NHO3ST@E7ACJ

**Comment by shakes76** [comment]
## TF/Torch Usage
Code was not running at time of submission was due -5
Solves the problem now working but changes after due date -3
See comments above from tutor
## Good Practice (Design/Commenting)
Good spacing and comments
No Header blocks -1
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present but not working on time -1
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages, could be more descriptive -1
Progressive commits used
## Documentation
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch)
The code did not work and will not be merged, but effort made to get it working, so feedback marks received.
Request Description OK
